freshmaker/handlers/koji/rebuild_eus_base_images.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2021  Red Hat, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import json
+import koji
+import re
+
+from freshmaker import conf, db
+from freshmaker.events import (
+    ErrataAdvisoryRPMsSignedEvent, ManualRebuildWithAdvisoryEvent)
+from freshmaker.handlers import ContainerBuildHandler, fail_event_on_handler_exception
+from freshmaker.lightblue import LightBlue
+from freshmaker.pulp import Pulp
+from freshmaker.errata import Errata
+from freshmaker.types import (
+    ArtifactType, ArtifactBuildState, EventState, RebuildReason)
+from freshmaker.models import Event, Compose
+from .rebuild_images_on_rpm_advisory_change import RebuildImagesOnRPMAdvisoryChange
+
+
+class RebuildEUSBaseImages(RebuildImagesOnRPMAdvisoryChange):
+    """
+    Rebuilds Extended Life Support base images
+    """
+
+    name = 'RebuildEUSBaseImages'
+
+    def can_handle(self, event):
+        if not super().can_handle(event):
+            return False
+
+        # If 'content_sets' of the event doesn't exist yet, create and fill it
+        if not hasattr(self.event, 'content_sets'):
+            errata = Errata()
+            errata_id = int(event.search_key)
+            # Use the errata_id to find out Pulp repository IDs from Errata Tool
+            # and furthermore get content_sets from Pulp where signed RPM will end
+            # up eventually when advisories are shipped.
+            pulp_repo_ids = list(set(errata.get_pulp_repository_ids(errata_id)))
```

Comment:
You can't call `_set_event_content_sets` instead of duplicating this code?

Reply:
Yeah you are right, probably forgot to replace it =)